src/main/java/ca/utoronto/fitbook/adapter/web/model/UserProfileModel.java

```diff
@@ -1,28 +1,37 @@
 package ca.utoronto.fitbook.adapter.web.model;
 
-import ca.utoronto.fitbook.entity.Exercise;
-import ca.utoronto.fitbook.entity.Post;
+import ca.utoronto.fitbook.application.port.out.response.ProfilePostResponse;
+import ca.utoronto.fitbook.application.port.out.response.UserProfileResponse;
 import lombok.NonNull;
 import lombok.Value;
 
 import java.util.List;
-import java.util.Map;
 
 @Value
 public class UserProfileModel {
     @NonNull
-    String id;
+    String profileId;
     @NonNull
     String name;
     int followingSize;
     int followerSize;
     @NonNull
     String joinDate;
     @NonNull
-    List<Post> postList;
+    List<ProfilePostResponse> postList;
     @NonNull
-    List<Post> likedPostList;
-    @NonNull
-    Map<String, Exercise> userExercises;
+    List<ProfilePostResponse> likedPostList;
     int totalLikes;
-}
+    public static UserProfileModel fromResponseToModel(UserProfileResponse response){
+        return new UserProfileModel(
+                response.getProfileId(),
+                response.getName(),
+                response.getFollowingSize(),
+                response.getFollowerSize(),
+                response.getJoinDate(),
+                response.getPostList(),
```

Comment:
You can use the following snipper to avoid using a for loop in the constructor in the UserProfileModel when ProfilePostModel class is created.
```suggestion
        postList.stream().map(ProfilePostModel::fromResponseToModel).collect(Collector.toList())
```